### DISCUSSION TITLE
XTTS pronounces period at end of sentences

### DISCUSSION THREAD

**Discussion by xdax1** [idiap/coqui-ai-TTS#163] [General]
Hello, I have a strange problem. I created a python script that creates a voiceover for subtitles from a .srt file. Sometimes in some issues the voiceover will add something more than is in the sentence, like “pontu” (subtitles are in Polish). I've tried many temperature and repetition_penalty settings, from the smallest to the largest, different voices but it's the same.
        “text_filtering": “standard”,
        “narrator_enabled": “false”,
        “language": “en”,
        “pitch": 0,
        “temperature": 0.75,
        “repetition_penalty": 10.0,
        “speed": 1.4,
I am using “apitts - xttsv2_2.0.3” from Alltalktts.

**Comment by eginhard** [comment]
Yes, this is a known issue with the XTTS model, see e.g. https://github.com/coqui-ai/TTS/issues/2952 and https://github.com/coqui-ai/TTS/issues/3445

**Comment by xdax1** [comment]
Only there it is only about “.” and in my case it also often says something else, sometimes longer even.
What temperature and repetition_penalty settings to make it not say anything additional best?

and another question by the way, it is possible to do so that, for example, the names read in English even though I have pl?

**Reply by eginhard** [reply]
Yes, these hallucinations can't be completely avoided with XTTS.

You could change the language token mid-sentence to do mixed-language TTS. Some foreign words it might also pronounce correctly out of the box already.
